src/components/fields/ObjectField.js

```diff
@@ -45,7 +45,7 @@ class ObjectField extends Component {
       onBlur,
     } = this.props;
     const { definitions, fields, formContext } = this.props.registry;
-    const { SchemaField, TitleField, DescriptionField } = fields;
+    const { ObjectPropertyField, TitleField, DescriptionField } = fields;
```

Comment:
With circular schema, we must avoid rendering everything: object properties should only be rendered when necessary.  I define a new component, `ObjectPropertyField` that only renders the underlying `SchemaField` when necessary—i.e. if the schema requires the property to be present, or if the `formData` contains an explicit value for it.